### PR TITLE
bugfix: #1072

### PR DESCRIPTION
**Pull request by ilanRosenbaum**
Add token to URL in analyzer so it can access private repos
Remove URL from logs to avoid token being in logs, replaced with what the URL would be without the token

### CODE REVIEW

File: source/plugins/languages/analyzer/analyzer.mjs [hunk 1]
```diff
@@ -86,21 +87,23 @@ export class Analyzer {
   /**Clone a repository */
   async clone(repository) {
     const {repo, branch, path} = this.parse(repository)
-    let url = /^https?:\/\//.test(repo) ? repo : `https://github.com/${repo}`
+    const token = core.getInput("token")
```
Comment (lowlighter) [inline source/plugins/languages/analyzer/analyzer.mjs:90]: Can you gate this behind a `if (process.env.GITHUB_ACTIONS)` so this only execute on GitHub Actions ?
The web instances probably doesn't need to expose its token on guest tokens (even though this indepth mode shouldn't be enabled on them anyways)

**Comment by lowlighter** [comment]
Hi 👋 !

Thanks for looking into this ! 
Just a small comment, else it looks good to me

The CI is probably broken so don't worry about it
I don't really have time to fix it urrently, especially since it'll be refactored in the v4 of metrics anyways

File: source/plugins/languages/analyzer/analyzer.mjs (hunk 1 above)
Comment (ilanRosenbaum) [reply source/plugins/languages/analyzer/analyzer.mjs]: Yes absolutely. Just pushed the change

**Comment by ilanRosenbaum** [comment]
Hey, any update on getting this merged? It's a small change and should make this product work better for a lot of users

**Review by lowlighter — APPROVED** [review]

**Comment by lowlighter** [comment]
Thanks for your contribution !

**Pull request merged**